machineconfiguration/v1alpha1/types_machineconfignode.go

```diff
@@ -0,0 +1,157 @@
+	// +kubebuilder:validation:Required
+	ConfigVersion MachineConfigNodeStatusMachineConfigVersion `json:"configVersion"`
+}
+
+// MachineConfigNodeStatusMachineConfigVersion holds the current and desired config versions as last updated in the MCN status.
+// When the current and desired versions are not matched, the machine config pool is processing an upgrade and the machine config node will
+// monitor the upgrade process.
+// When the current and desired versions are not match, the machine config node will ignore these events given that certain operations happen both during the MCO's upgrade mode and the daily operations mode.
+type MachineConfigNodeStatusMachineConfigVersion struct {
+	// current is the name of the machine config currently in use on the node.
+	// This value is updated once the machine config daemon has completed the update of the configuration for the node.
+	// This value should match the desired version unless an upgrade is in progress.
+	// Must be a lowercase RFC-1123 hostname (https://tools.ietf.org/html/rfc1123)
+	// +kubebuilder:validation:MaxLength=253
+	// +kubebuilder:validation:Pattern=`^([a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9\-]{0,61}[a-zA-Z0-9])(\.([a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9\-]{0,61}[a-zA-Z0-9]))*$`
+	// +optional
+	Current string `json:"current"`
+	// desired is the MachineConfig the node wants to upgrade to.
+	// The machine state controller sets this value once the MCO has generated a new MachineConfig for a node to upgrade to.
+	// Must be a lowercase RFC-1123 hostname (https://tools.ietf.org/html/rfc1123)
+	// +kubebuilder:validation:MaxLength=253
+	// +kubebuilder:validation:Pattern=`^([a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9\-]{0,61}[a-zA-Z0-9])(\.([a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9\-]{0,61}[a-zA-Z0-9]))*$`
+	// +kubebuilder:validation:Required
+	Desired string `json:"desired"`
+}
+
+// MachineConfigNodeSpecMachineConfigVersion holds the desired config version for the current observed machine config node.
+// When Current != Desired; the MachineConfigOperator is in an upgrade phase and the machine config node will
+// take account of upgrade related events. Otherwise they will be ignored given that certain operations
+// happen both during the MCO's upgrade mode and the daily operations mode.
+type MachineConfigNodeSpecMachineConfigVersion struct {
+	// desired is the name of the machine config that the the node should be upgraded to.
+	// This value is set when the machine config pool generates a new version of its rendered configuration.
+	// When this value is changed, the machine config daemon starts the node upgrade process.
+	// If the transition of this value is accepted, the new desired version will be updated in the .status.configVersion.desired field
+	// and the upgrade will proceed.
+	// Must be a lowercase RFC-1123 hostname (https://tools.ietf.org/html/rfc1123)
```

Comment:
```suggestion
	// Must be a lowercase RFC-1123 hostname (https://tools.ietf.org/html/rfc1123)
	// It may consist of only alphanumeric characters, hyphens (-) and period s(.) 
	// and must be at most 253 characters in length.
```

Reply:
applied